### DISCUSSION TITLE
Send data from client to server in Next.js especially in AppDir

### DISCUSSION THREAD

**Discussion by ErAz7** [vercel/next.js#42835] [App Router]
Is there any way to pass some data from client-side to server components
This can be very helpful for proper Auth implementation and even in some other cases.
Imagine I want to fetch user profile on all pages (server-side) but with an important condition, necessary for acceptable performance:

> the profile should not be fetched if it's already present on client

But the problem is neither in Next 12 nor in AppDir I cant find any way to send such data (at lest some boolean flags) to server so server cannot have any idea about when to fetch profile

**Comment by ErAz7** [comment]
after thinking bit more on it, doesn't make much sense to have such implementation since server delay for fetching profile will be very small.
actually a fake 1sec delay on express server causing loading state on all pages made me think of this implementation

**Comment by icyJoseph** [comment]
I don't think it is possible. 

One could think of client side selector/text-input which onChange or submission, updates the prop it passes to it's child server-component. Let me clear here, **AFAIK this is not possible**. 

In general the question/problem is, can we import Server Components into Client Components? It doesn't seem like it.

Instead what they recommend is to use composition, https://beta.nextjs.org/docs/rendering/server-and-client-components#importing-server-components-into-client-components

However, then we lose the ability to pass client side props to the server component.

In your case though, I am not sure, it sounds like something you should handle with cookies rather than manually sending payloads yourself.

**Comment by harshilAmbliya** [comment]
i have same review on it . because i try hard to find it but i dont get anything but in authentication of it i find js-cookie pacake which can store cookies from client side and handle middleware also
